tests/integration/upgrade/test_sharding_rollback.py

```diff
@@ -102,36 +101,54 @@ async def test_rollback_on_config_server(
         shard_two_actual_writes == shard_two_expected_writes["number"]
     ), "continuous writes to shard two failed during upgrade"
 
+    await ops_test.model.wait_for_idle(
+        apps=CLUSTER_COMPONENTS, status="active", timeout=1000, idle_period=20
+    )
+
     # after all shards have upgraded, verify that the balancer has been turned back on
     # TODO implement this check once we have implemented the post-cluster-upgrade code DPE-4143
 
 
 @pytest.mark.runner(["self-hosted", "linux", "X64", "jammy", "large"])
 @pytest.mark.group(1)
 @pytest.mark.abort_on_fail
-@pytest.mark.skip("Need a new version published with upgrade bug fixed")
 async def test_rollback_on_shard_and_config_server(
     ops_test: OpsTest, continuous_writes_to_shard_one, continuous_writes_to_shard_two
 ) -> None:
     """Verify that a config-server and shard can safely rollback without losing writes."""
     new_charm = await ops_test.build_charm(".")
     await run_upgrade_sequence(ops_test, CONFIG_SERVER_APP_NAME, new_charm=new_charm)
 
-    shard_unit = await find_unit(ops_test, leader=True, app_name=SHARD_ONE_APP_NAME)
-    action = await shard_unit.run_action("pre-upgrade-check")
-    await action.wait()
-    assert action.status == "completed", "pre-upgrade-check failed, expected to succeed."
```

Comment:
Yeah exactly.
This part was replicating the `run_upgrade_sequence` but without calling the function, so better do with the neat function instead of code duplication.